src/Microsoft.OData.Core/ODataMessageWriterSettings.cs

```diff
@@ -346,6 +382,79 @@ internal SelectedPropertiesNode SelectedProperties
             }
         }
 
+        /// <summary>
+        /// Gets or sets a value that indicates whether the writer should put key values in their own URI segment when automatically building URIs.
+        /// If this value is false, automatically-generated URLs will take the form "../EntitySet('KeyValue')/..".
+        /// If this value is true, automatically-generated URLs will take the form "../EntitySet/KeyValue/..".
+        /// This setting only applies to URLs that are automatically generated by the <see cref="ODataMessageWriter" /> and does not modify URLs explicitly provided by the user.
+        /// </summary>
+        public bool EnableWritingKeyAsSegment { get; set; }
+
+        /// <summary>
+        /// Get whether to write OData control information without a prefix
+        /// True if control information can be read without prefix 'odata.', otherwise false.
+        /// The default value is false for OData 4.0 and true for OData 4.01.
+        /// The option is applied during deserialization.
+        /// </summary>
+        /// <returns>Whether to omit the OData prefix for the specified version.</returns>
+        public bool GetOmitODataPrefix()
+        {
+            return this.enableWritingODataAnnotationWithoutPrefix;
+        }
+
+        /// <summary>
+        /// Version-specific behavior for writing OData control information without a prefix
+        /// True if control information can be read without prefix 'odata.', otherwise false.
+        /// The default value is false for OData 4.0 and true for OData 4.01.
+        /// The option is applied during deserialization.
+        /// </summary>
+        /// <param name="version">The version of the version-specific behavior being requested.</param>
+        /// <returns>Whether to omit the OData prefix for the specified version.</returns>
+        public bool GetOmitODataPrefix(ODataVersion version)
+        {
+            if (version >= ODataVersion.V401)
+            {
+                return this.omitODataPrefix;
+            }
+
+            return this.omitODataPrefix40;
+        }
+
+
+        /// <summary>
+        /// Whether to write OData control information without a prefix
+        /// True to read control information without the prefix 'odata.', otherwise false.
+        /// The default value is false for OData 4.0 and true for OData 4.01.
+        /// The option is applied during deserialization.
+        /// </summary>
+        /// <param name="enabled">True to omit writing the OData prefix, False to write the prefix.</param>
+        public void SetOmitODataPrefix(bool enabled)
+        {
+            this.enableWritingODataAnnotationWithoutPrefix =
+            this.omitODataPrefix =
+            this.omitODataPrefix40 = enabled;

```

Comment:
This methods sets the value for both 4.0 and 4.01. Is that the same behaviour as before?

Reply:
This is how this method implemented in `ODataSimplifiedOptions`. I'm not sure whether changing the behaviour is a good idea unless we first assess the impact of doing so. Also, there's an overload of this method that takes the ODataVersion as argument.

Reply:
@habbes You can retain the logic as is but at least indicate to the user that there exists an overload that lets them specify the OData version